bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/BagVerifier.scala

```diff
@@ -200,3 +200,35 @@ trait BagVerifier[BagLocation <: Location, BagPrefix <: Prefix[BagLocation]]
         )
     }
 }
+
+trait UnpackedBagVerifier[BagLocation <: Location, BagPrefix <: Prefix[BagLocation]]
+    extends BagVerifier[BagLocation, BagPrefix] {
+  def verifyBag(
+    ingestId: IngestID,
+    root: BagPrefix,
+    space: StorageSpace,
+    externalIdentifier: ExternalIdentifier
+  ): Try[IngestStepResult[VerificationSummary]] =
+    verifyStandaloneBag(
+      ingestId = ingestId,
+      root = root,
+      space = space,
+      externalIdentifier = externalIdentifier
+    )
+}
+
+trait ReplicatedBagVerifier[BagLocation <: Location, BagPrefix <: Prefix[BagLocation]]
+    extends BagVerifier[BagLocation, BagPrefix] {
+  def verifyBag(
+    ingestId: IngestID,
+    root: BagPrefix,
+    space: StorageSpace,
+    externalIdentifier: ExternalIdentifier
+  ): Try[IngestStepResult[VerificationSummary]] =
+    verifyStandaloneBag(
+      ingestId = ingestId,
+      root = root,
+      space = space,
+      externalIdentifier = externalIdentifier
+    )
```

Comment:
The idea is that eventually this `verifyBag` method would take an extra argument `srcBagRoot`, and it would compare the `tagmanifest-sha256.txt` file before checking the checksums in the actual bag.

Because the function signatures are eventually going to diverge, that means divergence in `Main` and in `BagVerifierWorker`, and it's worth discussing how we do that before going on further.